weave/trace/vals.py

```diff
@@ -780,10 +781,16 @@ def make_trace_obj(
                 )
             )
             val = from_json(read_res.obj.val, project_id, server)
-        except ObjectDeletedError as e:
-            # encountered a deleted object, return DeletedRef, warn and continue
-            val = DeletedRef(ref=new_ref, deleted_at=e.deleted_at, error=e)
-            logger.warning(f"Could not read deleted object: {new_ref}")
+        except httpx.HTTPStatusError as e:
```

Comment:
the fact that `server.obj_read` raises an `httpx.HTTPStatusError` exception is an implementation detail of the trace server itself and the way that the remotehttptraceserver works. I'd recommend either:
1. putting this try catch inside the remotehttptraceserver and returning a synthetic objectschema with `deleted_at`; or
2. raising a specific error in the server that can be caught here.

Reply:
okay bad choice to use `httpx.HTTPStatusError`, but this is basically option 2 right? The trace server is raising a specific error, that we are trying to identify here. We can blanket catch `Exception` and do the existing logic, I just thought that was less specific than at least narrowing the exception to an http exception. I do understand what you are saying though. 

Option 1 feels even worse tbh, i'm not 100% sure I understand the philosophical purpose of remotehttptraceserver but I think I would be surprised if I saw type conversion to a synthetic object schema in there. I guess if it's purpose is as broad as general middlewear to process from the request layer down to our client it's appropriate...